### DISCUSSION TITLE
Provided List<Json>, expected *UpdateManyMutationInput

### DISCUSSION THREAD

**Discussion by pranavanmaru** [prisma/prisma#14466] [Q&A] [answered]
Bulk update call fails with an error on types. 
```prisma
type UserUpdateManyMutationInput {
  id?: String | StringFieldUpdateOperationsInput
  sId?: String | StringFieldUpdateOperationsInput
  FirstName?: String | NullableStringFieldUpdateOperationsInput | Null
  LastName?: String | NullableStringFieldUpdateOperationsInput | Null
  Email?: String | NullableStringFieldUpdateOperationsInput | Null
  Telephone?: String | NullableStringFieldUpdateOperationsInput | Null
}
```
> *Error*
[...
 {
    id: 'a60105e3-632f-4e6e-ac10-ca6165dda55f',
    sId: '1000000017327501',
    FirstName: 'Three',
    LastName: 'And-Only',
    Email: 'three@only.com',
    Telephone: '',
  }]
on prisma.updateManynonprofitsUser. Provided List<Json>, expected nonprofitsUserUpdateManyMutationInput or nonprofitsUserUncheckedUpdateManyInput

Despite being virtually the same, on attempting to bulk update the User table i'm seeing the above error. 
Note: createMany for the same type works with no issues. 

Here's the bulk update call:
```ts
        await prisma.User.updateMany({
          data: updateUsersChunk,
        });
```
`updateUsersChunk` here is an array of objects shown in the error message above.
what am I doing wrong here?

**Comment by austincrim** [comment, marked as answer]
Hey @pranavanmaru 👋,

Could you share your Prisma schema?

**Reply by pranavanmaru** [reply]
``` prisma 
type User {
  id                                                                               String                         @id(map: "PK__user__3213E83F97BC02DA") @default(dbgenerated("newid()"), map: "DF__User__id__0114F6F8") @db.UniqueIdentifier
  sId                                                                              String                         @unique(map: "AK_USER_ALTERNATE_KEY") @db.NVarChar(25)
  FirstName                                                                        String?                        @db.NVarChar(255)
  LastName                                                                         String?                        @db.NVarChar(255)
  Email                                                                            String?                        @db.NVarChar(255)
  Telephone                                                                        String?                        @db.NVarChar(100)
}
```

**Reply by pranavanmaru** [reply]
@austincrim it's almost identical to the type it throws in the error. I've added the schema above.

**Reply by austincrim** [reply]
`updateMany` accepts a single object for the `data` option, not an array of objects. Its purpose is to "update many" rows with the same data, so passing an array of objects does not work in that context.

Could you describe what you are trying to accomplish a bit more?

**Reply by pranavanmaru** [reply]
@austincrim I was attempting to do a bulk update using `updateMany` since its counterpart `createMany` seemed to be working for bulk insertions. docs suggest the `where` attribute in `updateMany` is optional which lead me to think `updateMany`, like the name suggests could be used to do bulk update in scenarios like data synchronizations.

**Reply by pranavanmaru** [reply]
thanks for the explanation. I'll use raw queries.

**Reply by ru-van-urk** [reply]
@pranavanmaru 

Do you have an example on how you achieved this with raw queries? I am very curious about how you did this.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
